### DISCUSSION TITLE
Genetic algorithm generation limit when using pyswmm

### DISCUSSION THREAD

**Discussion by zsbeheshti** [pyswmm/pyswmm#450] [Q&A]
Dear all,
I use the genetic algorithm for optimization and use pyswmm to extract the time series of Head and Flow. But to run this code, the genetic algorithm accepts a limited number of generations, while there is no such limit for other codes that do not use pyswmm. Does anyone have any related experience to share with me?
Regards

**Comment by bemcdonnell** [comment]
@zsbeheshti, could you please elaborate on what limit there is? Also, adding some code for your workflow could be helpful. Maybe it’s a number of files open problem? Hard to say unless I see the code.  I have a ton of experience using GA.

**Reply by zsbeheshti** [reply]
It is not possible to choose a high number for generation and population. For example, in my code, a maximum of 5 generations and 180 populations can be set, and if it is more than that, the following two errors will be received:
FileExistsError: [WinError 183] Cannot create a file when that file already exists
ERROR 307: cannot open binary results file.

**Reply by bemcdonnell** [reply]
@zsbeheshti Can you share the code you are using to specifically open and close the OUT file?  I am wondering if it is being closed properly.

**Reply by bemcdonnell** [reply]
Also, when you use pyswmm.Simulation() to launch a sim, are you using unique file names for the RPT/OUT for Generation and Individual?
